extensions/INTEL/SPV_INTEL_maximum_registers.asciidoc

```diff
@@ -0,0 +1,195 @@
+- Greg Lueck, Intel +
+// spell-checker: enable
+
+== Notice
+
+Copyright (c) 2024 Intel Corporation.  All rights reserved.
+
+== Status
+
+* Complete
+
+== Version
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | {docdate}
+| Revision           | 1
+|========================================
+
+== Dependencies
+
+This extension is written against the SPIR-V Specification, Version 1.6 Revision
+3.
+
+This extension requires SPIR-V 1.0.
+
+== Overview
+
+This extension adds an execution mode to specify the maximum number of registers
+a SPIR-V consumer should use when compiling an entry point.
+This is a hint only that does not modify the functional behavior of the program,
+but can change its performance characteristics.
+
+== Extension Name
+
+To use this extension within a SPIR-V module, the appropriate *OpExtension* must
+be present in the module:
+
+[subs="attributes"]
+----
+OpExtension "SPV_INTEL_maximum_registers"
+----
+
+== Modifications to the SPIR-V Specification, Version 1.6
+
+=== Validation Rules
+
+Add validation rules to section 2.16.1 Universal Validation Rules under Entry Point:
+
+* Each *OpEntryPoint* must contain at most one of the
+*MaximumRegistersINTEL*, *MaximumRegistersIdINTEL*, or
+*NamedMaximumRegistersINTEL* execution modes.
+
+=== Capabilities
+
+Modify Section 3.31, Capability, adding rows to the Capability table:
+
+--
+[cols="^.^1,5,3",options="header"]
+|====
+2+^| Capability ^| Implicitly Declares 
+| {capability_token} | *RegisterLimitsINTEL* +
+Specifies the maximum number of registers that may be used by an entry point.
+|
+|====
+--
+
+=== Execution Modes
+
+Modify Section 3.6, Execution Mode, adding rows to the Execution Mode table:
+
+[cols="^4,20,3*5,22",options="header",width = "100%"]
+|====
+2+^.^| Execution Mode 3+<.^| Extra Operands 
+| Enabling Capabilities
+
+| {maximum_registers_literal_token} | *MaximumRegistersINTEL* +
+Specifies the maximum number of registers to be allocated to a single
+invocation.
+This is a performance hint only.
+If the specified number of registers is not supported then the compiler may
+choose the closest number of supported registers or may ignore the request.
+3+| 'Literal' +
+'Number of Registers`
+| *RegisterLimitsINTEL*
+
+| {maximum_registers_id_token} | *MaximumRegistersIdINTEL* +
+Same as the *MaximumRegistersINTEL* execution mode but using an '<id>'
+operand instead of a literal.
+The operand must be an integer type scalar and is interpreted as an unsigned
+value.
+3+| '<id>' +
+'Number of Registers`
+| *RegisterLimitsINTEL*
+
+| {named_maximum_registers_token} | *NamedMaximumRegistersINTEL* +
+Specifies the maximum number of registers to be allocated to a single invocation
+using a named policy rather than a specific numeric number of registers.
+This is a performance hint only.
+If the named policy is not supported then the compiler may ignore the request.
+3+| 'Literal' +
+'Named Maximum Number of Registers'
```

Comment:
See discussion: https://github.com/KhronosGroup/SPIRV-Headers/pull/416#discussion_r1483075943

We may want to change this to a specific "kind" rather than "LiteralInteger", something like:

```suggestion
3+| 'Named Maximum Number of Registers' +
'Named Maximum Number of Registers'
```

Reminder: if we accept this suggestion then we need to regenerate the HTML spec!

Reply:
looks good